python-app/openapi-diff/action.yaml

```diff
@@ -0,0 +1,98 @@
+name: OpenApi Diff
+description: Generate and comment OpenAPI Diff
+
+inputs:
+  base_ref:
+    description: "Base branch for the diff"
+    required: true
+    default: "main"
+  head_ref:
+    description: "Head branch for the diff"
+    required: true
+
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: set up python
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.11"
+
+    - uses: actions/checkout@v3
+      with:
+        ref: "${{github.base_ref}}"
+
+    - name: upgrade dependencies
+      run: |
+        python -m pip install --upgrade pip pipenv
+      shell: bash
+
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        ref: "${{github.event.inputs.base_ref}}"
+
+    - name: install dependencies
+      run: pipenv sync
+      shell: bash
+
+    - name: Generate OpenAPI for base
+      run: |
+        pipenv run openapi
+        mkdir specs
+        cp doc/api-spec.yaml ./specs/base-spec.yaml
```

Comment:
Maybe this path should be an argument so it adapts to different repos with different paths and names